client/wsclient.go

```diff
@@ -242,26 +308,29 @@ func (c *wsClient) runOneCycle(ctx context.Context) {
 	)
 	r.ReceiverLoop(ctx)
 
+	// If we exited receiverLoop it means there is a connection error or the connection
+	// has closed. We cannot read messages anymore, so clean up the connection.
+	// If there is a connection error we will need to start over.
+
 	// Stop the background processors.
 	procCancel()
 
-	// If we exited receiverLoop it means there is a connection error, we cannot
-	// read messages anymore. We need to start over.
+	// Wait for WSSender to stop.
+	c.sender.WaitToStop()
 
-	// Close the connection to unblock the WSSender as well.
+	// Close the connection.
 	_ = c.conn.Close()
```

Comment:
I am not sure about the swapping of the order of "close" vs "wait". The comment "Close the connection to unblock the WSSender as well" seems to indicate closing is necessary to ensure wait complete. Is that the case or the comment was misleading?

Reply:
I may have misunderstood it, but to me the comment was misleading. Closing the connection doesn't appear to have any effect on the WSSender since the only way to shut down WSSender is by [canceling the context](https://github.com/open-telemetry/opamp-go/blob/main/client/internal/wssender.go#L59), which is done by calling `procCancel` in this function.

I mostly reversed these for two reasons:
1. This allows us to clean up the application-level side of the connection before shutting down the transport-level component, which feels like the proper order of operations to me.
2. In the case where we had an error on receiving (e.g. we got bad proto data) and we're looking to reset the connection, this will allow us to send any remaining messages before closing it.